cosmos/models/Workflow.py

```diff
@@ -560,10 +562,21 @@ def _process_finished_tasks(jobmanager):
     for task in jobmanager.get_finished_tasks():
         if task.NOOP or task.exit_status == 0:
             task.status = TaskStatus.successful
-            yield task
+        elif task.scheduling_error:
```

Comment:
Good point on the infinite retries. Let me think about that. As for merging -- I want to have this stuff have a nightly build or two under its belt before we do that. Thanks for your feedback, it's been incredibly helpful